File: library/rxobservablelist/src/main/java/com/github/mayurkaul/rxobservablelist/RxBaseObject.java
package com.github.mayurkaul.rxobservablelist;

import io.reactivex.subjects.BehaviorSubject;
import io.reactivex.subjects.Subject;

public abstract class RxBaseObject<T> {

    private BehaviorSubject<STATUS> mStatusObservable  = BehaviorSubject.create();

    abstract void setKey(T key);

    abstract T getKey();

    void setStatus(STATUS status)
    {
        mStatusObservable.onNext(status);
    }

    Subject<STATUS> getStatusObservable()
    {
        return mStatusObservable;
    }

    public enum STATUS
    {
        ADDED,UPDATED,REMOVED
    }

}

File: library/rxobservablelist/src/main/java/com/github/mayurkaul/rxobservablelist/RxBaseObject.kt
package com.github.mayurkaul.rxobservablelist

import io.reactivex.subjects.BehaviorSubject
import io.reactivex.subjects.Subject

internal abstract class RxBaseObject<T> {

    private var mStatusObservable: BehaviorSubject<STATUS> = BehaviorSubject.create<STATUS>();

    abstract fun setKey(key: T)

    abstract fun getKey(): T

    fun setStatus(status: STATUS)
    {
        mStatusObservable.onNext(status);
    }

    fun getStatusObservable():Subject<STATUS>
    {
        return mStatusObservable;
    }

    enum class STATUS
    {
        ADDED,UPDATED,REMOVED
    }

}
